Multi-omic Applications (#17)
greenelab/knowledge-graph-review: content/04.network_applications.md

### Unifying Applications

Knowledge graphs have been used in many biomedical applications ranging from identifying protein functions [@doi:10.1186/s12859-018-2163-9] to prioritizing cancer genes [@doi:10.1093/bioinformatics/bty148] to recommending safer drugs to patients [@arxiv:1710.05980; @doi:10.1609/aaai.v33i01.33011126] (Figure {@fig:unifying_applications}).
In this section we discuss how knowledge graphs are being applied in biomedical settings. 
We put particular emphasis on an emerging set of techniques: those that project knowledge graphs into a low dimensional space.

![
Overview of biomedical applications that make use of knowledge graphs.
Categories consist of: (a) Multi-Omic applications, (b) Pharmaceutical Applications and (c) Clinical Applications.
](images/figures/unifying_applications_overview.png){#fig:unifying_applications}

#### Multi-Omic Applications

> Multi-omic applications consist of studying genes (genomics), RNA (transcriptomics) and proteins (proteomics).

Reviewer: ```suggestion
Multi-omic applications for knowledge graphs include efforts to study the genome, how genes are expressed in the transcriptome, and how the products of those transcripts interact in the proteome.
```

> Approaches in this category use knowledge graphs to establish connections between biological entities (genes, RNA, proteins) themselves and diseases via tasks such as gene prioritization [@doi:10.1093/jamia/ocy117], predicting interactions between proteins [@doi:10.1089/cmb.2012.0273;10.1186/1752-0509-9-S1-S9], and detecting MiRNA-disease associations [@doi:10.1155/2017/2498957].

Reviewer: ```suggestion
Approaches in this category use knowledge graphs to establish connections between -omic entities as well as diseases.
Such tasks include gene-symptom prioritization [@doi:10.1093/jamia/ocy117], protein-protein interaction prediction [@doi:10.1089/cmb.2012.0273;10.1186/1752-0509-9-S1-S9], and detecting miRNA-disease associations [@doi:10.1155/2017/2498957].
```

> In this section we discuss on multi-omic applications that utilize projecting knowledge graphs into a low dimensional space to make connections.

Reviewer: ```suggestion
We focus specifically on multi-omic applications of algorithms that project knowledge graphs into a low dimensional space to make connections.
```

Knowledge graphs have been used as recommendation systems to establish links between RNA with disease and proteins with other proteins.
Shen et al. [@doi:10.1155/2017/2498957] used an algorithm called collaborative filtering to establish an association between MiRNA and diseases.

> The authors constructed an MiRNA-Disease network using the HMDDD database [@doi:10.1093/nar/gky1010] and generated an adjacency matrix with the rows representing MiRNA and the columns representing diseases.

Reviewer: ```suggestion
The authors constructed an miRNA-Disease network using the HMDDD database [@doi:10.1093/nar/gky1010] and generated an adjacency matrix with the rows representing miRNA and the columns representing diseases.
```

> This adjacency matrix was decomposed into small rectangular matrices using SVD, then these matricies were used to calculate similarity scores between the MiRNA and diseases.

Reviewer: It is hard to use the github interface to make each change. MiRNA should be miRNA. The proper noun is lower cased because the portmanteau of micro RNA was initially done that way.

> High scores implicated a high likelihood that a given MiRNA has an association with a given disease [@doi:10.1155/2017/2498957].

Reviewer: ```suggestion
High scores implied a high likelihood that a given MiRNA had an association with a given disease [@doi:10.1155/2017/2498957].
```

Other approaches have built off of Shen et al.'s [@doi:10.1155/2017/2498957] work by incorpoating novel ways to perform matrix factorization [@doi:10.3390/genes10020080; @doi:10.1186/s12859-019-2956-5;@doi:10.1186/s12859-019-3260-0] or by integrating machine learning models in conjunction with matrix factorization [@doi:10.1109/TCBB.2019.2937774].

> These methods provided high AUROCs, but new discoveries were hard to validate as there is limited experimental data [@doi:10.1155/2017/2498957].

Reviewer: ```suggestion
These methods provided high AUROCs, but new discoveries have been hard to validate as experiments in this space are costly and time consuming at best [@doi:10.1155/2017/2498957].
```

Apart from MiRNA, collaborative filtering has been used to predict protein-protein interactions [@doi:10.1109/BIBM.2010.5706537; @doi:10.1089/cmb.2012.0273; @doi:10.1186/1752-0509-9-S1-S9].

> Future work could include doing a blinded literature search to increase validation power or incorporate other techniques as deep learning to increase detection power.

Reviewer: ```suggestion
Though extensive validation of newly generated candidates may be impractical, it would be helpful to see future efforts in this space include a blinded literature search for prioritized and randomly selected candidates as part of the standard evaluation of such approaches.
```

> Approaches that use deep learning have mainly focused on using the node2vec model [@arxiv:1607.00653] or variants of that model.

Reviewer: ```suggestion
Approaches that use deep learning have mainly used the node2vec model [@arxiv:1607.00653] or variants of that model.
```

The generated knowledge graph was embedded via node2vec and similarity scores were calculated for every gene-symptom pair in the graph.
High scores implicated high chance for an association [@doi:10.1093/jamia/ocy117].
This approach outperformed methods that didn't use a knowledge graph; however, validation was difficult as it involved manual curation of the literature @doi:10.1093/jamia/ocy117].
Similar approaches used variants of the node2vec algorithm to predict gene-disease associations
[@doi:10.1093/bioinformatics/bty559; doi:10.3389/fgene.2019.00226 @doi:10.1186/s12920-019-0627-z; @doi:10.1109/BIBM47256.2019.8983134] analyze RNA-seq data [@doi:10.1093/nar/gkx750] and infer novel protein information [@doi:10.1093/bioinformatics/btx275; @doi:10.1007/978-3-030-00825-3_16; @doi:10.1016/j.artmed.2019.04.001; @doi:10.1186/s12859-018-2163-9].
Future extensions of these applications should consider incorporating more sources of data such as compounds, anatomic locations or even gene pathways to improve prediciton performance.

Knowledge graphs have aided the multi-omics field by generating novel discoveries.

> Majority of methods have been heavily focused on matrix factorization and node2vec to project knowledge graph into a low dimensional space; however, translational models remain an untapped resource that may benefit the field in future applications.

Reviewer: ```suggestion
Most approaches to date use matrix factorization and node2vec to project knowledge graph into a low dimensional space, and translational models may be an untapped resource that could aid future efforts.
```

> Also, another area of exploration is to incorporate heterogeneous information such as anatomic locations or genetic pathways into generate networks to improve novel discoveries

Reviewer: ```suggestion
Another area of exploration could be the incorporation of multiple sources of information such as anatomic locations or genetic pathways to improve the specificity of findings (i.e., to predict that a protein-protein interaction happens in a specific cell type or tissue).
```
